packages/web/src/apollo/links.tsx

```diff
@@ -123,11 +128,11 @@ export type RedwoodApolloLink<
   link: Link
 }
 
-export type RedwoodApolloLinks = [
-  RedwoodApolloLink<'withToken'>,
-  RedwoodApolloLink<'authMiddleware'>,
-  RedwoodApolloLink<'enhanceErrorLink'>,
-  RedwoodApolloLink<'httpLink', HttpLink>
-]
+export type RedwoodApolloLinks = Array<
+  | RedwoodApolloLink<'withToken'>
+  | RedwoodApolloLink<'authMiddleware'>
+  | RedwoodApolloLink<'enhanceErrorLink'>
+  | RedwoodApolloLink<'httpLink', HttpLink>
+>
```

Comment:
```suggestion
export type RedwoodApolloLinks = Array<
  | RedwoodApolloLink<Exclude<RedwoodApolloLinkName, 'httpLink'>>
  | RedwoodApolloLink<'httpLink', HttpLink>
>
```

Reply:
I thought that was the intention of the type though? To "lock" what links can be introduced without changing the type? I don't really like types this specific, was just trying to maintain the convention.

Reply:
You still have to update the `RedwoodApolloLinkName` type. It's just to make it less likely the two goes out of sync.

Actually, now that I type it out, this should probably be the source of truth. So yeah, let's leave this as it is, and see if we can figure out a way to construct `RedwoodApolloLinkName` from the `RedwoodApolloLinks` type

```
export type RedwoodApolloLinks = Array<
  | RedwoodApolloLink<'withToken'>
  | RedwoodApolloLink<'authMiddleware'>
  | RedwoodApolloLink<'enhanceErrorLink'>
  | RedwoodApolloLink<'httpLink', HttpLink>
>

export type RedwoodApolloLinkName = /* Value of first generic type passed to `RedwoodApolloLink` in `RedwoodApolloLinks */
```